### PR TITLE
Advanced hardsuit gets advanceder

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Chief engineers hardsuit's helmet now automatically deploys in hazardous environment when smartmode is enabled, its a verb under the hardsuit and starts disabled

**Changelog:**
:cl:
add: Chief engineer's advanced hardsuit can automatically deploy in hazardous environments
/:cl:

### CODE REVIEW

**Comment by Citinited** [comment]
Advanced hardsuit gets advancederer

**Comment by Couls** [comment]
This could also be changed to apply to all hardsuits that have attachable helmets but I think its most fitting for the CE's hardsuit

**Comment by Fox-McCloud** [comment]
I really don't think this is necessary to be honest.

There's better ways to make it advanced, like the integrated jetpack system TG has.

**Comment by taukausanake** [comment]
Or actually having RIG suits instead? I forget what happened to those

**Comment by Couls** [comment]
I don't think its necessary either, I just think it'd be a neat feature and just adds to the futuristic atmosphere

**Comment by MrMagolor** [comment]
> Or actually having RIG suits instead? I forget what happened to those

What is dead may never die

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -388,27 +388,57 @@
...
 	if(status_flags & GODMODE)	return 1	//godmode
-
+	if(istype(wear_suit,/obj/item/clothing/suit/space/hardsuit/elite))
+		H = wear_suit
 	if(adjusted_pressure >= dna.species.hazard_high_pressure)
 		if(!(HEATRES in mutations))
 			var/pressure_damage = min( ( (adjusted_pressure / dna.species.hazard_high_pressure) -1 )*PRESSURE_DAMAGE_COEFFICIENT , MAX_HIGH_PRESSURE_DAMAGE)
 			take_overall_damage(brute=pressure_damage, updating_health = TRUE, used_weapon = "High Pressure")
 			throw_alert("pressure", /obj/screen/alert/highpressure, 2)
+			if (H && H.helmet && !head)
```
Comment (variableundefined) [inline code/modules/mob/living/carbon/human/life.dm]: Spacing please

**Comment by variableundefined** [comment]
I like the feature in theory, however the implementation is not ideal. This implementation make it so that everytime someone is in a high pressure area, they're checked whether for an advanced hardsuit and whether to deploy it. This presents two problems:

1. Ideally, item behaviour should be attached to the item, not a seemingly irrelevant segment of life code meant for taking damage and alarm. This make for less maintainable code.
2. This is very inefficient. Every single person, every tick they're in a high pressure area, is being checked for an advanced hardsuit. But only one person will be wearing it at any time.

If this is going to be implemented it should be attached to the suit in a process proc, instead of the alert screen. However this is not a very useful buff for the hardsuit as it will only apply to newbie.

So I cannot support the PR.

**Comment by variableundefined** [comment]
Closing for reasons stated above.

**Pull request closed without merging**